### PR TITLE
DM-35624: Create a task in analysis_tools to measure per-visit metrics

### PR DESCRIPTION
**Pull request by jeffcarlin**
In addition to the 5-sigma limiting magnitude calculation, I also added sky source statistics. For the sky stats, I moved them into their own standalone file containing both the objectTable and sourceTable versions.

### CODE REVIEW

**Review by fred3m — APPROVED** [review]
Looks good overall. A few minor comments, mostly due to potentially unnecessary overrides and a method that I think was removed on accident.

File: python/lsst/analysis/tools/interfaces.py [hunk 1]
```diff
@@ -306,23 +306,6 @@ class AnalysisTool(AnalysisAction):
...
-        Parameters
-        ----------
-        context : `Context`
-            The specific execution context, this may be a single context or
-            a joint context, see `Context` for more info.
-        """
-        self.prep.applyContext(context)
-        self.process.applyContext(context)
-        self.produce.applyContext(context)
-
```
[the remark this answers is not in the document]
Comment (jeffcarlin) [reply python/lsst/analysis/tools/interfaces.py:325]: Actually, that was intentional. Nate realized while we were working through the Context implementation that this code wasn't needed.

File: python/lsst/analysis/tools/actions/vector/selectors.py [hunk 2]
```diff
@@ -132,11 +132,35 @@ def setDefaults(self):
...
 
 
+class VisitPlotFlagSelector(FlagSelector):
+    def getInputSchema(self) -> KeyedDataSchema:
+        yield from super().getInputSchema()
+
+    def __call__(self, data: KeyedData, **kwargs) -> Vector:
+        result: Optional[Vector] = None
+        temp = super().__call__(data, **kwargs)
+        if result is not None:
```
[the remark this answers is not in the document]
Comment (jeffcarlin) [reply python/lsst/analysis/tools/actions/vector/selectors.py:142]: I think this is a question for @natelust, as it seems to be a general choice.

File: python/lsst/analysis/tools/actions/vector/selectors.py [hunk 3]
```diff
@@ -132,11 +132,35 @@ def setDefaults(self):
         self.selectWhenTrue = ["detect_isPatchInner", "detect_isDeblendedSource"]
 
 
+class VisitPlotFlagSelector(FlagSelector):
+    def getInputSchema(self) -> KeyedDataSchema:
+        yield from super().getInputSchema()
```
[the remark this answers is not in the document]
Comment (jeffcarlin) [reply python/lsst/analysis/tools/actions/vector/selectors.py:137]: I also think @natelust would have a much better answer than I could muster.

**Pull request merged**